### PR TITLE
Render live demo box iframe

### PR DESCRIPTION
**Pull request by minkyngkm**
## Done

- Added functionality to Live demo box
- Set default example in the JSON file

Fixes [#1205](https://github.com/canonical-web-and-design/vanilla-squad/issues/1205)

## QA

- Open [demo](https://vanilla-framework-4322.demos.haus/docs/patterns/notification)
- Check the initial default "information" example loads as expected
- Change values of all selects/switches and check the example works as expected

### CODE REVIEW

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            <div className="col-2">
+              <Select id="style" label="Style" name="style" onChange={handleChange} options={configValues.dropdown.style}/>
+            </div>
+          </div>
+          <div className="row" style={{gridGap: 0}}>
+            <div className="p-card col-6">
+              <h3>Example</h3>
+              <p className="p-card__content">
+              <div>
+                <div dangerouslySetInnerHTML={iframe()} />
```
Comment (carkod) [inline templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:78]: Does this not work with `<iframe src=${url} width="100%" height="450"></iframe>` directly?

**Comment by carkod** [comment]
I was also doing this one, but I see that you've got the state and the types right also, so I'm going to close my PR. A couple of things:
- This PR is pointing to main, you should change it to `canonical-web-and-design: live-demo-dox`
- Also left a comment about the way you are rendering the iframe

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx (hunk 1 above)
Comment (minkyngkm) [reply templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:78]: Oh it directly works! Thanks @carkod!🙏🏻

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx [hunk 2]
```diff
@@ -34,28 +40,42 @@ const LiveDemoBox = () => {
...
+              <Select id="style" label="Style" name="style" onChange={handleChange} options={configValues.dropdown.style}/>
             </div>
           </div>
           <div className="row" style={{gridGap: 0}}>
             <div className="p-card col-6">
               <h3>Example</h3>
-              <p className="p-card__content">space for exmaple here</p>
+              <p className="p-card__content">
+              <div>
+                <iframe src={url} width="100%" height="200"></iframe>
```
Comment (carkod) [inline templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:73]: I believe the original `example.js` calculates the height based on the size of the element? If it's fixed at 200, elements that have more than this height will require scrolling

**Comment by mrgnr** [comment]
@minkyngkm No worries if you still have refactoring or other changes you want to make, but would you please run Prettier before merging? You can do this by running `yarn run format-js` from the repository root. Thanks :slightly_smiling_face:

**Comment by minkyngkm** [comment]
> 

@bethcollins92 and I were still refactoring. Yep I will make sure to rebase and run Prettier before merging! Thanks @mrgnr for your great work! 🙌

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx (hunk 2 above)
Comment (minkyngkm) [reply templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx]: Thanks @carkod for your review! Yep I will update the height 👍

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx [hunk 3]
```diff
@@ -18,44 +18,121 @@ type InitialState = {
...
+type SelectValueType = {
+  [key: string]: string | boolean;
+};
+
 const LiveDemoBox = () => {
   const [configValues, setConfigValues] = useState<InitialState>();
+  const [selectValue, setSelectValues] = useState<SelectValueType>({
+    type: "",
+    theme: "",
+    style: "",
```
Comment (mrgnr) [inline templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:30]: A minor detail, but I think that `dismiss`, `actions` and `timestamp` should be included here as well.

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx [hunk 4]
```diff
@@ -18,44 +18,121 @@ type InitialState = {
...
+              />
             </div>
           </div>
-          <div className="row" style={{gridGap: 0}}>
+          <div className="row" style={{ gridGap: 0 }}>
             <div className="p-card col-6">
               <h3>Example</h3>
-              <p className="p-card__content">space for exmaple here</p>
+              <p className="p-card__content">
+                <div>
```
Comment (mrgnr) [inline templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:126]: I see an error in the console that says `<div> cannot appear as a descendant of <p>`. I think we can just change the `p` to `div`.

**Review by mrgnr — CHANGES_REQUESTED** [review]
I saw some errors in the console about missing keys for list items (React wants you to explicitly add a `key` attribute whenever you have a list or generate items using `map`). This can be fixed by changing `query` to `key` and adding a `key` attribute to each item in the Switch component. I made a commit [here](https://github.com/mrgnr/vanilla-framework/commit/dd30ff0b5ff9c57d42d17868e5de57e905a22ed6) if you just want to cherry-pick it into this branch :)

File: templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx (hunk 3 above)
Comment (bethcollins92) [reply templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:30]: We weren't sure how to do this, as the values of these are going to be different depending on the variant. type theme and style will always be consistent, but the other values will be different

Comment (mrgnr) [reply templates/static/js/src/demobox/components/LiveDemoBox/LiveDemoBox.tsx:30]: That makes sense, it should be fine to leave it as it is

**Review by mrgnr — APPROVED** [review]
Looks good! :rocket:

**Pull request merged**